include/bcs/HagelaarEnergyAdvectionBC.h

```diff
@@ -24,23 +24,24 @@ class HagelaarEnergyAdvectionBC : public ADIntegratedBC
 
   Real _r_units;
   Real _r;
-
+  const unsigned int _num_ions;
   // Coupled variables
 
   const ADVariableGradient & _grad_potential;
-  MooseVariable & _ip_var;
-  const ADVariableValue & _ip;
-  const ADVariableGradient & _grad_ip;
-
-  const MaterialProperty<Real> & _sgnip;
-  const ADMaterialProperty<Real> & _muip;
-  const ADMaterialProperty<Real> & _Dip;
-  const MaterialProperty<Real> & _se_coeff;
-  const MaterialProperty<Real> & _se_energy;
+  std::vector<MooseVariable *> _ip_var;
```

Comment:
I believe that I did this simply because there were some other bcs that had already been implemented and did the same but I can make the change in other places I saw it as well.

Reply:
After doing some digging the reason that this was changes to a pointer instead of a reference is that you are unable to create a vector of references directly. If we want to do this we will need to use the `std::reference_wrapper` and I can make that change as well if you want @cticenhour

Reply:
Thanks for linking the StackOverflow discussion; I didn't know this nuance of vectors. I am OK with this change (and others like it).